src/libs/TransactionUtils/index.ts

```diff
@@ -817,6 +818,26 @@ function getRateID(transaction: OnyxInputOrEntry<Transaction>): string | undefin
     return transaction?.modifiedCustomUnitRateID ?? transaction?.comment?.customUnit?.customUnitRateID?.toString();
 }
 
+/**
+ * Whether expense requests connected with given transaction IDs are unsubmitted by the owners
+ */
+function areRequestsInTransactionsUnsubmitted(transactionIDs: string[]): boolean {
+    const allReports = ReportConnection.getAllReports();
+
+    const reports = transactionIDs.flatMap((transactionID) => {
+        const selectedTransaction = getTransaction(transactionID);
+        const selectedReport = ReportUtils.getReport(selectedTransaction?.reportID ?? '-1');
+
+        return Object.values(allReports ?? {}).filter((report) => report?.parentReportID === selectedReport?.reportID ?? '-1');
+    });
+
+    return reports.some((report) => {
+        const parentReportAction = ReportActionsUtils.getReportAction(report?.parentReportID ?? '-1', report?.parentReportActionID ?? '-1');
+
+        return ReportActionsUtils.isMoneyRequestAction(parentReportAction) && !ReportUtils.canEditMoneyRequest(parentReportAction) && ReportUtils.isReportInGroupPolicy(report);
```

Comment:
Does it make sense to move these line to a new method? So that `HoldReasonPage.tsx` can also reuse it and it'll be easier to maintain how to decide whether a report can be hold or not. See

https://github.com/Expensify/App/blob/d4d5a2586910ff46147219ee3e98bb3e936f8037/src/pages/iou/HoldReasonPage.tsx#L41-L44

https://github.com/Expensify/App/blob/d4d5a2586910ff46147219ee3e98bb3e936f8037/src/pages/iou/HoldReasonPage.tsx#L50

https://github.com/Expensify/App/blob/d4d5a2586910ff46147219ee3e98bb3e936f8037/src/pages/iou/HoldReasonPage.tsx#L68

Reply:
Agreed, I've moved the redundant code and put it in `ReportUtils` for ease of use across multiple components.